src/Raven.Server/Web/Operations/OperationsHandler.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System.Net;
+using System.Threading.Tasks;
+using Raven.Server.Documents;
+using Raven.Server.Routing;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
+
+namespace Raven.Server.Web.Operations
+{
+    public class OperationsHandler : DatabaseRequestHandler
+    {
+
+        [RavenAction("/databases/*/operations/next-operation-id", "GET")]
+        public Task GetNextOperationId()
```

Comment:
If we go with single (and probably very long) request, then:
1. How should we handle progress? If progress should be send via response, then we have to use websockets. jQuery can't handle chunked response - it only returns XHR after request is completed. Hopefully browsers has separate pools for web sockets connections and we shouldn't have issues with choking request pool. 
2. How someone else can track progress of this operation? It is technically possible by having running task on server side and we can use /operations/get-all endpoint. We can do pulling (using auto-refresh widget like in v3.5) or use changes api. 
3. Do we care when request is disconnected ? When having changes API we can easily reconnect and watch for progress, completion status. 
4. We also have to return operation id in response to be able to kill this operation. (I'm not sure if we can cancel running ajax request and server will be notified about this). 
5. We also have to check what happens when client closes tab/browser (connection about on server side?)